pkg/events/derive/derive.go

```diff
@@ -0,0 +1,78 @@
+package derive
+
+import (
+	"fmt"
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// DeriveFunction is a function prototype for a function that receives an event as
+// argument and may produce a new event if relevant.
+// It returns a derived or empty event, depending on successful derivation,
+// and an error if one occurred.
+type DeriveFunction func(trace.Event) ([]trace.Event, []error)
+
+// EventSkeleton is a struct for the necessary information from an event definition to create the derived event
+type EventSkeleton struct {
```

Comment:
So if you just keep `derive` inside the events package I don't think you need all of this.
Just use the `event` struct instead as the skeleton, because that literally is what the event struct is.